src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const sumObject = [];
+  let newState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(newState, action.extraData);
+        break;
+
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => {
+          delete newState[key];
+        });
+        break;
+
+      case 'clear':
+        newState = {};
+        break;
+    }
+
+    sumObject.push({ ...newState });
```

Comment:
The operation of creating a copy of `newState` is repeated in every case of the switch statement. To adhere to the DRY principle, you should move this operation outside of the switch statement, performing it only once after the switch block.

Reply:
so the tests fail if you make such a change